packages/cli/src/ActiveExecutions.ts

```diff
@@ -73,10 +75,13 @@ export class ActiveExecutions {
 				fullExecutionData.workflowId = workflowId;
 			}
 
-			executionId = await this.executionRepository.createNewExecution(fullExecutionData);
-			if (executionId === undefined) {
-				throw new ApplicationError('There was an issue assigning an execution id to the execution');
-			}
+			executionId = await Db.transaction(async (transactionManager) => {
```

Comment:
why not put the transaction in the repository? IMO all database related code be inside the repository classes.

Reply:
Yeah it should probably be there. At the moment we have some issues in how we use repositories that makes them a bit ambiguous about how they should be used:

1. Repositories should be a contract between domain layer objects and data persistence objects. We don't really make distinction between domain layer and data layer, and use the same objects in most of the places
2. The contract should be explicit, and more in the domain's language. At the moment we don't define the contract, but use whatever method is available at the typeorm's base repositories (e.g. `userRepository.save` instead of `userRepository.update`
3. Using transactions with repositories is currently cumbersome. Our repositories are singletons in the DI container. We need to pass the transaction as an optional extra parameter to the repository functions if we want to execute that in the scope of the transaction. Ideally it should be clear where the transaction boundary is and who should be responsible for initiating transaction.

I'm open to any ideas how we could improve the situation. I can also try to think of ways to improve it.

But for this case I'll move it to the repository